### PR TITLE
chore: cleanup simple typings

### PR DESCRIPTION
**Pull request by dorianj**
### Summary of Changes

no functional changes: just fix "easy" typing changes, i.e. those where author's intent was clear, but the right guards or type declarations were in place.

### Tests

Some tests were changed to use option chaining. Might be a bit confusing to get a `null` or `undefined`, but I think the correct thing rather than writing a bunch of detailed assertions

### Documentation

n/a

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/common/EditableText/index.tsx [hunk 1]
```diff
@@ -90,11 +88,11 @@ class EditableText extends React.Component<
   }
 
   exitEditMode = () => {
-    this.props.setEditMode(false);
+    this.props.setEditMode?.(false);
```
Comment (Golodhros) [inline amundsen_application/static/js/components/common/EditableText/index.tsx:91]: Oh, interesting, I didn't thought to use it like this! 
TIL!

**Review by Golodhros — APPROVED** [review]
LGTM

Thanks a lot for doing this! It made our code much better and move us forward a strictly typed codebase.

**Comment by dorianj** [comment]
@Golodhros Pushed to fix the DCO (i gotta figure out how to get that to automatically get added). I don't have write bits, so could you please merge?

**Comment by Golodhros** [comment]
Thanks again Dorian!

**Pull request merged**